### PR TITLE
ENH: Allow Series.apply to accept list-like and dict-like

### LINKED ISSUE #39140: ENH: Allow apply/agg to accept list-like and dict-like
```
s = pd.Series([1, 2])
df = pd.DataFrame([1, 2])

# Reducers: list-like
ops = np.array([np.sum, np.mean])
print(df.agg(ops))
print(s.agg(ops))
print(df.apply(ops))
print(s.apply(ops))  # Raises

# Reducers: dict-like
ops = pd.Series({0: np.mean})
print(df.agg(ops))  # Raises
print(s.agg(ops))  # Raises
print(df.apply(ops))  # Raises
print(s.apply(ops))  # Raises

# Transforms - list-like
ops = np.array([np.abs, np.sqrt])
print(df.transform(ops))
print(df.apply(ops))
print(s.transform(ops))
print(s.apply(ops))  # Raises

# Transforms - dict-like
ops = pd.Series({0: np.abs})
print(df.transform(ops))
print(s.transform(ops))
print(s.apply(ops))  # Raises
print(df.apply(ops))  # Raises
```

All lines that don't raise give the expected output.

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #39140
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

I only added the Series.apply list/dict-like addition to the whatsnew. For both Series and DataFrame .agg, this PR also enables passing a Series as a dict-like, but other dict-likes already work. It didn't seem notable to add this to the whatsnew.

### CODE REVIEW

File: pandas/core/apply.py [hunk 1]
```diff
@@ -195,6 +195,24 @@ def maybe_apply_str(self) -> Optional[FrameOrSeriesUnion]:
...
+
+        Returns
+        -------
+        result: Series, DataFrame, or None
+            Result when self.f is a list-like or dict-like, None otherwise.
+        """
+        # Note: dict-likes are list-like
+        if not is_list_like(self.f):
+            return None
+        # pandas\core\apply.py:144: error: "aggregate" of "DataFrame" gets
```
Comment (jreback) [inline pandas/core/apply.py:210]: ```**{axis':self.axis, **self.kwds})```  would work here

**Comment by jreback** [comment]
if my comment works ok, otherwise lgtm. merge either way.

**Comment by rhshadrach** [comment]
Thanks @jreback - your comment did work, but I found that just removing the keyword `axis` also works.

**Comment by jreback** [comment]
thanks @rhshadrach

**Pull request merged**